src/TextField/TextFieldLabel.js

```diff
@@ -35,13 +43,30 @@ export default function TextFieldLabel(props, context) {
   }, classNameProp);
 
   return (
-    <label className={className} {...other} />
+    <label className={className} {...other}>
+      {children}
+      {required ? <span className={asteriskClassName}>{'\u2009'}*</span> : null}
```

Comment:
Isn't the left operator of `&&` always casted to something that is bool-ish? `"foo" && 42 => 42`, `{ strange: "object" } && { anotherStrange: "thing" } => { anotherStrange: "thing" }`